nilearn/experimental/surface/_surface_image.py

```diff
@@ -169,13 +190,110 @@ class SurfaceImage:
 
     def __init__(
         self,
-        mesh: PolyMesh | dict[str, Mesh],
-        data: PolyData | dict[str, Mesh],
+        mesh: PolyMesh | dict[str, Mesh | str | Path],
+        data: (
+            PolyData | dict[str, Mesh | str | Path] | Nifti1Image | str | Path
+        ),
     ) -> None:
+        """Create a SurfaceImage instance.
+
+        Parameters
+        ----------
+        mesh : PolyMesh | dict[str, Mesh  |  str  |  Path]
+        data : PolyData | dict[str, Mesh  |  str  |  Path] | Niimg-like object
+        """
         self.mesh = mesh if isinstance(mesh, PolyMesh) else PolyMesh(**mesh)
-        self.data = data if isinstance(data, PolyData) else PolyData(**data)
+
+        if not isinstance(data, (PolyData, dict, str, Path, Nifti1Image)):
+            raise TypeError(
+                "'data' must be one of"
+                "[PolyData, dict, str, Path, Nifti1Image].\n"
+                f"Got {type(data)}"
+            )
+
+        if isinstance(data, PolyData):
+            self.data = data
+        elif isinstance(data, dict):
+            self.data = PolyData(**data)
+        elif isinstance(data, (Nifti1Image, str, Path)):
+            self._vol_to_surf(data)
+
         _check_data_and_mesh_compat(self.mesh, self.data)
+
         self.shape = self.data.shape
 
     def __repr__(self) -> str:
         return f"<{self.__class__.__name__} {getattr(self, 'shape', '')}>"
+
+    def _vol_to_surf(self, img: Nifti1Image | str | Path, **kwargs) -> None:
+        """Extract surface data from a Nifti image.
```

Comment:
keeping as a private function for now

Reply:
Well, it's more than an extraction, it's a projection.